xonsh/aliases.py

```diff
@@ -254,7 +254,8 @@ def source_cmd(args, stdin=None):
     args[0] = fpath if fpath else args[0]
     if not os.path.isfile(args[0]):
         raise FileNotFoundError(args[0])
-    prevcmd = 'call "{}"\necho off'.format(args[0])
+    prevcmd = 'call "{}"'.format('" "'.join(args))
```

Comment:
@BYK I was thinking it was up to the user to escape their commands if they were using `source-cmd`.

On the other hand it may not be what people expect. Also Popen normally takes care of escaping the individual arguments, but we can't use that here here since the whole command is written to a batch file. Thus we have to use the whole procedure from your link not only the cmd.exe specific ^escapes.